client/src/components/Overview.jsx

```diff
@@ -1,41 +1,87 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
 
 import ProductDetails from './ProductDetails.jsx'
 import ImageGallery from './ImageGallery.jsx'
 
 import { makeStyles } from '@material-ui/core/styles';
+import CircularProgress from '@material-ui/core/CircularProgress';
 import Grid from '@material-ui/core/Grid';
 import Typography from '@material-ui/core/Typography';
 
 const useStyles = makeStyles({
   root: {
-    border: "dotted 1px grey",
+    // border: "dotted 1px grey", //component outline
   },
 
   prodOverview: {
     padding: "20px",
   }
 });
 
-const Overview = (props) => {
+
+const Overview = ({ currentProduct }) => {
   const classes = useStyles();
+  const [productStyles, setProductStyles] = useState([]);
+  const [styleIndex, setStyleIndex] = useState(0);
 
-  return (
-      <Grid className={classes.root} container direction="row" justifyContent="flex-start" alignItems="flex-start" >
-        <Grid item xs={7}>
-          <ImageGallery />
-        </Grid>
-        <Grid item xs={5}>
-          <ProductDetails />
-        </Grid>
+  let productId = currentProduct.id
+
+  useEffect(() => {
+    axios.get(`/api/products/${productId}/styles`)
+    .then(({ data }) => {
+      setProductStyles(data.results)
+    }).catch((err) => {
+      console.log('Unable to retireve product overview details')
+    });
+  }, [productId])
```

Comment:
I think you can put an empty array as the second argument. The current product changes when the product state of the App component changes. The App component will render again with the new product and pass that down to this Overview component so it doesn't really need to watch for changes to the productId. (I could be wrong, but either way it's not a big deal)

Reply:
iI only want the useEffect to run when the product ID changes, otherwise, updating state directly will will only change the details that are being displayed, but not make extraneous get requests to update that that will most likely stay the same.